pennylane/transforms/dynamic_one_shot.py

```diff
@@ -287,6 +307,40 @@ def measurement_with_no_shots(measurement):
     return tuple(normalized_meas) if len(normalized_meas) > 1 else normalized_meas[0]
 
 
+def gather_mcm_jit(circuit_measurement, measurement, is_valid):
+    """Combines, gathers and normalizes several measurements with trivial measurement values
+    when the Catalyst compiler is active.
+
+    Args:
+        circuit_measurement (MeasurementProcess): measurement
+        measurement (TensorLike): measurement results
+        samples (List[dict]): Mid-circuit measurement samples
```

Comment:
This docstring does not match the function arguments. Also, in my opinion, `circuit_measurement` should be `measurement`, and `measurement` should be `mcm_samples`.